src/core/qgsmapcanvasinterface.h

```diff
@@ -0,0 +1,43 @@
+/***************************************************************************
+                         qgsmapcanvasinterface.h
+                         ------------------
+    begin                : November 2018
+    copyright            : (C) 2018 by Denis Rouzaud
+    email                : denis@opengis.ch
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSMAPCANVASINTERFACE_H
+#define QGSMAPCANVASINTERFACE_H
+
+#include "qgis_core.h"
+#include "qgis_sip.h"
+
+class QgsMapSettings;
+
+/**
+ * \ingroup core
+ * The QgsMapCanvasInterface class provides a simple interface to access
+ * core components of a map canvas.
+ * \since QGIS 3.6
+ */
+class CORE_EXPORT QgsMapCanvasInterface
```

Comment:
I don't like the naming of this class -- it's dangerously bleeding gui specific classes into core. What about something more generic like "QgsMapDisplayInterface" or "QgsRenderedMapInterface"? QgsLayoutItemMap could implement this interface too.